wordpress/wp-content/plugins/sil-dictionary-webonary/webonary/Webonary_Cloud.php

```diff
@@ -0,0 +1,225 @@
+<?php
+
+class Webonary_Cloud
+{
+	public static $doBrowseByLetter = 'browse';
+
+	public static $doGetEntry = 'get';
+
+	public static $doSearchFulltext = 'search';
+
+	private static function isValidEntry($entry) {
+		return is_object($entry) && isset($entry->_id);
+	} 
+
+	private static function convertGuidToId($guid) {
+		return 'g' . $guid;
+	} 
+
+	private static function remoteGet($request, $args = array()) {
+		if (!defined('WEBONARY_CLOUD_API_URL'))  {
+			error_log('WEBONARY_CLOUD_API_URL is not set! Please do so in wp-config.php.');
+			return;
+		}
+
+		$url = WEBONARY_CLOUD_API_URL . $request;
```

Comment:
How is this concatenation always going to produce a valid URL? Is it guaranteed there will be a slash (/) in the right places?

Reply:
filter_var($url, FILTER_VALIDATE_URL)